spec.html

```diff
@@ -7680,6 +7680,99 @@ <h1>Jobs and Host Operations to Enqueue Jobs</h1>
     </emu-note>
 
+    <emu-clause id="sec-jobcallback-records">
+      <h1>JobCallback Records</h1>
+      <p>The JobCallback is a Record value used to store a callable object and a host-defined ambient context value. Callable objects that are invoked via a Job enqueued by the host may have additional host-defined context. To propagate the state, Job Abstract Closures should not capture and call callable objects directly. Instead, use MakeJobCallback and CallJobCallback.</p>
+      <p>The host hooks HostGetJobCallbackAmbientContext and HostSetJobCallbackAmbientContext control the ambient context.</p>
+      <emu-note>
+        <p>For example, WHATWG HTML uses the ambient context to propagate the incumbent settings object for Promise callbacks.</p>
+      </emu-note>
+      <p>JobCallback Records have the fields listed in <emu-xref href="#table-jobcallback-records"></emu-xref>.</p>
+      <emu-table id="table-jobcallback-records" caption="JobCallback Record Fields">
+        <table>
+          <tbody>
+            <tr>
+              <th>
+                Field Name
+              </th>
+              <th>
+                Value
+              </th>
+              <th>
+                Meaning
+              </th>
+            </tr>
+            <tr>
+              <td>
+                [[Callback]]
+              </td>
+              <td>
+                A callable object, or *undefined*
+              </td>
+              <td>
+                The callback to invoke when the Job is invoked.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                [[HostAmbientContext]]
+              </td>
+              <td>
+                Any
+              </td>
+              <td>
+                Field reserved for use by hosts that associate additional context with callbacks invoked from Jobs.
+              </td>
+            </tr>
+          </tbody>
+        </table>
+      </emu-table>
+    </emu-clause>
+
+    <emu-clause id="sec-hostgetjobcallbackambientcontext" aoid="HostGetJobCallbackAmbientContext">
+      <h1>HostGetJobCallbackAmbientContext ( )</h1>
+      <p>HostGetJobCallbackAmbientContext is a host-defined abstract operation that allows hosts to provide an ambient context to associate with a callback. This is called at the time that the callback is passed to the function that is responsible for its being eventually scheduled and run.</p>
+      <p>The implementation of HostGetJobCallbackAmbientContext must conform to the following requirements:</p>
+      <ul>
+        <li>It must always complete normally (i.e., not return an abrupt completion).</li>
+      </ul>
+      <p>The default implementation of HostGetJobCallbackAmbientContext returns *undefined*.</p>
```

Comment:
```suggestion
      <p>The default implementation of HostGetJobCallbackAmbientContext returns ~empty~.</p>
```
? also not sure if it matters

Reply:
I reworked the PR and this host hook was removed.